runtime/src/message_processor.rs

```diff
@@ -722,10 +734,16 @@ impl MessageProcessor {
         rent_collector: &RentCollector,
         log_collector: Option<Rc<LogCollector>>,
         executors: Rc<RefCell<Executors>>,
+        mut instruction_recorders: Option<&mut Vec<InstructionRecorder>>,
         cluster_type: ClusterType,
         epoch: Epoch,
     ) -> Result<(), TransactionError> {
         for (instruction_index, instruction) in message.instructions.iter().enumerate() {
+            let instruction_recorder = instruction_recorders.as_mut().map(|recorders| {
```

Comment:
could these be constructed as a single shot earlier (in the bank) and then looked up here via `instruction_index`?

Reply:
Ah yeah, that's nicer. PR here: https://github.com/solana-labs/solana/pull/12467